zipline/assets/assets.py

```diff
@@ -809,6 +882,79 @@ def lookup_future_symbol(self, symbol):
             raise SymbolNotFound(symbol=symbol)
         return self.retrieve_asset(data['sid'])
 
+    def lookup_by_supplementary_mapping(self, mapping_type, value, as_of_date):
+        try:
+            owners = self.supplementary_map[
```

Comment:
Oh interesting, hadn't thought of that. I wonder if the hit we take for loading everything in is worth avoiding the added complexity? Guess it depends how many different types of mappings you have...